class.jetpack.php

```diff
@@ -568,28 +568,20 @@ static function delete_active_modules() {
 	public function add_configure_hook() {
 		global $wp_filter;
 
-		$current_priority = has_filter( 'plugins_loaded', array( $this, 'configure' ) );
-		if ( false !== $current_priority ) {
-			remove_action( 'plugins_loaded', array( $this, 'configure' ), $current_priority );
+		if ( false === has_filter( 'plugins_loaded', [ $this, 'configure' ] ) ) {
+			add_action( 'plugins_loaded', [ $this, 'configure' ], - PHP_INT_MAX );
```

Comment:
While this seems to be a good way to secure the top spot in the callback sequence, I'd rather us all stay away from `PHP_INT_MAX`. Tinkering with it, I have come upon several "interesting" irregularities around it that I wouldn't want to have in my code:
* With the introduction of `PHP_INT_MIN` constant in PHP 5.6 there's a new way to set the minimal possible integer, and it's different from `- PHP_INT_MAX` by exactly 1. This means you're not running at the maximum priority anyway.
* All arithmetic operations with numbers around the max or min integer are quirky. For example, trying to decrease the minimum number by one yields the same minimum number, whereas multiplying it yields zero. I wouldn't want my plugin to rely on numeric indexes that behave in such a weird way around simple calculations.
* Even if you're confident that you have not missed anything from the previous two quirks and you're still running at the highest priority, one can use floats to overcome that. Just set your callback's priority to `-10E+20`, and you're before everything that's at `PHP_INT_MIN`;

My point here is should we instead find a way to stay one higher than the rest? Or if you really need to be the first, maybe you should be using the `plugin_loaded` hook for your plugin?

Reply:
@zinigor, as I mentioned in the [issue thread ](https://github.com/Automattic/jetpack/issues/14757#issuecomment-589752400), we won't change our priorities in a near future.

You are right about `PHP_INT_MIN`. I think we started using `PHP_INT_MAX` quite a long time ago and when we switched to PHP 5.6, we did not re-evaluate our priorities. But so far we didn't have noticeable issues so we prefer not to change it.

The [WP documentation](https://developer.wordpress.org/reference/functions/add_action/) says that the priority should be an integer, so using `-PHP_INT_MAX` (or even `PHP_INT_MIN`) is still valid, while using a float is does not comply with the API (and I believe it can behave in a weird way).

As I said before, we won't change our priority for now on WPML, and we are legitimate to keep it as is.
 
That's why I am suggestion Jetpack to immediately take the higher priority so there's no need to change it over and over. And it will also simplify the code, we won't have to have an edge case when the number it already  `PHP_INT_MIN`.